### PR TITLE
Added typescript to all components

### PR DESCRIPTION
**Pull request by pi0trpietruszewski**
Added typescript to library, added bob the builder, refactored code

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -35,6 +35,6 @@ src/
 ## Code Style
 * Make sure you are using linter with linting rules defined in ESLint config (.eslinrc)
 * Name branch according to your ticket following this pattern: RNS-XX-short_description
-* Imports and exports inside `index.js` files eg. `screens/index.js`, `components/index.js` should be alfabetically
+* Imports and exports inside `index.tsx` files eg. `screens/index.tsx`, `components/index.tsx` should be alfabetically
```
Comment (IdaszakDaniel) [inline docs/CONTRIBUTING.md:38]:
```suggestion
* Imports and exports inside `index.tsx` files eg. `screens/index.tsx`, `components/index.tsx` should be sorted alfabetically
```

File: src/StickyParallaxHeader.tsx [hunk 2]
```diff
@@ -66,44 +149,49 @@ class StickyParallaxHeader extends Component {
...
             : scrollHeight;
         const scrollNode = getSafelyScrollNode(this.scroll);
-        scrollNode.scrollTo({ y: scrollTargetPosition, duration: 1000 });
+        scrollNode?.scrollTo({ y: scrollTargetPosition });
       }, 250);
     }
   }
 
   componentWillUnmount() {
+    // https://reactnative.dev/docs/animatedvaluexy#removealllisteners
```
Comment (IdaszakDaniel) [inline src/StickyParallaxHeader.tsx:158]: Is this comment necessary?

File: src/StickyParallaxHeader.tsx [hunk 3]
```diff
@@ -66,44 +149,49 @@ class StickyParallaxHeader extends Component {
...
     const snapToEdgeThreshold = snapStartThreshold || height / 2;
     const id = snapToEdgeAnimatedValue.addListener((value) => {
-      scrollNode.scrollTo({ x: 0, y: value.y, animated: false });
+      scrollNode?.scrollTo({ x: 0, y: value.y, animated: false });
     });
-
-    if (y < -20 && !constants.isAndroid && !refreshControl) this.spring(y);
+    //@TODO Check that
+    if (y < -20 && !constants.isAndroid && !refreshControl) this.spring();
+    // if (y < -20 && !constants.isAndroid && !refreshControl) this.spring(y);
```
Comment (IdaszakDaniel) [inline src/StickyParallaxHeader.tsx:194]: Are you able to remove "TODO" and comments?

File: src/StickyParallaxHeader.tsx [hunk 4]
```diff
@@ -314,11 +399,11 @@ class StickyParallaxHeader extends Component {
       onMomentumScrollEnd,
       onMomentumScrollBegin,
     } = this.props;
-    const { currentPage, isFolded } = this.state;
+    const { currentPage /*, isFolded */ } = this.state;
```
Comment (IdaszakDaniel) [inline src/StickyParallaxHeader.tsx:402]: Can we remove `isFolded`?

File: src/StickyParallaxHeader.tsx [hunk 5]
```diff
@@ -403,22 +489,18 @@ class StickyParallaxHeader extends Component {
...
             keyboardShouldPersistTaps={keyboardShouldPersistTaps}>
             {!tabs && children}
             {tabs &&
               tabs.map((item) => (
                 <View
-                  tabLabel={item.title}
+                  accessibilityLabel={item.title}
                   key={item.title}
-                  onLayout={this.setContentHeight}
+                  // onLayout={this.setContentHeight}
```
Comment (IdaszakDaniel) [inline src/StickyParallaxHeader.tsx:503]: Can we remove this comment?

File: src/components/ScrollableTabView/ScrollableTabView.tsx [hunk 6]
```diff
@@ -1,33 +1,63 @@
...
+  scrollEnabled: boolean;
+};
+type SceneKeys = string[];
+type State = {
+  currentPage: number;
+  scrollXIOS: Animated.Value;
+  containerWidth: number;
+  sceneKeys: SceneKeys;
+};
+class ScrollableTabView extends React.Component<ScrollableTabViewProps, State> {
```
Comment (IdaszakDaniel) [inline src/components/ScrollableTabView/ScrollableTabView.tsx:44]: add empty line

File: src/components/ScrollableTabView/ScrollableTabView.tsx [hunk 7]
```diff
@@ -77,34 +105,42 @@ class ScrollableTabView extends React.Component {
...
+     * @TODO Check that block of code
+     */
     if (typeof localNextPage === 'object') {
-      localNextPage = nextPage.nativeEvent.position;
+      // @ts-ignore
+      localNextPage = nextPage?.nativeEvent?.position ?? 0;
     }
+    /**
+     * End
+     */
```
Comment (IdaszakDaniel) [inline src/components/ScrollableTabView/ScrollableTabView.tsx:120]: Can we remove it?

File: src/predefinedComponents/AvatarHeader/AvatarHeader.tsx [hunk 8]
```diff
@@ -110,21 +121,21 @@ class AvatarHeader extends React.Component {
             hitSlop={sizes.hitSlop}
             onPress={rightTopIconOnPress}
             style={styles.rightHeaderButton}>
+            {/*@ts-ignore There is default props for that*/}
```
Comment (IdaszakDaniel) [inline src/predefinedComponents/AvatarHeader/AvatarHeader.tsx:124]: are those ts-ignores necessary?

File: src/predefinedComponents/AvatarHeader/AvatarHeader.tsx [hunk 9]
```diff
@@ -248,93 +259,58 @@ class AvatarHeader extends React.Component {
...
-          header={this.renderHeader()}
-          deviceWidth={constants.deviceWidth}
           scrollEvent={event([{ nativeEvent: { contentOffset: { y: this.scrollY.y } } }], {
             useNativeDriver: false,
-            listener: (e) => scrollEvent && scrollEvent(e),
+            listener: (e: NativeSyntheticEvent<NativeScrollEvent>) => scrollEvent && scrollEvent(e),
           })}
-          headerSize={this.setHeaderSize}
-          headerHeight={headerHeight}
+          // deviceWidth={constants.deviceWidth}
```
Comment (IdaszakDaniel) [inline src/predefinedComponents/AvatarHeader/AvatarHeader.tsx:266]: Should we remove it? Maybe it was a regression where prop was not passed at all?

File: src/predefinedComponents/DetailsHeader/DetailsHeader.tsx [hunk 10]
```diff
@@ -20,25 +46,26 @@ class DetailsHeader extends React.Component {
...
 
-  renderHeader = (user) => {
+  renderHeader = (user: DetailsData): ReactElement => {
     const {
       backgroundColor,
       leftTopIconOnPress,
       rightTopIconOnPress,
       leftTopIcon,
       rightTopIcon,
+      //Without this comment eslint breaks Prettier
```
Comment (IdaszakDaniel) [inline src/predefinedComponents/DetailsHeader/DetailsHeader.tsx:68]: 🤔

File: src/StickyParallaxHeader.tsx (hunk 2 above)
Comment (pi0trpietruszewski) [reply src/StickyParallaxHeader.tsx]: It explains why we have !ts-ignore, I will add additional comment for that

File: src/StickyParallaxHeader.tsx (hunk 3 above)
Comment (pi0trpietruszewski) [reply src/StickyParallaxHeader.tsx]: Done

File: src/StickyParallaxHeader.tsx (hunk 4 above)
Comment (pi0trpietruszewski) [reply src/StickyParallaxHeader.tsx]: 👍

File: src/components/ScrollableTabView/ScrollableTabView.tsx (hunk 6 above)
Comment (pi0trpietruszewski) [reply src/components/ScrollableTabView/ScrollableTabView.tsx:44]: 👍

File: src/StickyParallaxHeader.tsx (hunk 5 above)
Comment (pi0trpietruszewski) [reply src/StickyParallaxHeader.tsx]: 👍

File: src/components/ScrollableTabView/ScrollableTabView.tsx (hunk 7 above)
Comment (pi0trpietruszewski) [reply src/components/ScrollableTabView/ScrollableTabView.tsx:120]: We need to check that part of code in future

File: src/predefinedComponents/AvatarHeader/AvatarHeader.tsx (hunk 8 above)
Comment (pi0trpietruszewski) [reply src/predefinedComponents/AvatarHeader/AvatarHeader.tsx:126]: Yes, somehow typescript won't recognise `static defaultProps` field :/

File: src/predefinedComponents/AvatarHeader/AvatarHeader.tsx (hunk 9 above)
Comment (pi0trpietruszewski) [reply src/predefinedComponents/AvatarHeader/AvatarHeader.tsx:268]: `deviceWidth` is not used anywhere in code, while refactoring using typescript i didn't found any usage of `deviceWidth` and in all components prop was missing in prop types.

File: src/predefinedComponents/DetailsHeader/DetailsHeader.tsx (hunk 10 above)
Comment (pi0trpietruszewski) [reply src/predefinedComponents/DetailsHeader/DetailsHeader.tsx:68]: Somehow the issue is gone

**Review by davidwico — APPROVED** [review]

File: src/predefinedComponents/components/UserModal/UserModal.tsx [hunk 11]
```diff
@@ -223,6 +246,7 @@ class UserModal extends React.Component {
         <Text style={styles.contentText}>{title}</Text>
         {cards.map((card) => (
           <QuizListElement
+            pressUser={() => {}}
```
[the remark this answers is not in the document]
Comment (pi0trpietruszewski) [reply src/predefinedComponents/components/UserModal/UserModal.tsx:249]: 👍

**Review by TomaszRej — APPROVED** [review]

**Review by IdaszakDaniel — APPROVED** [review]

**Pull request merged**